src/utils/applicationinsights.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * Application insights client configuration.
+ *
+ * To use this aside winston:
+ *
+ * Set APPINSIGHTS_INSTRUMENTATIONKEY environment variable
+ * then add a new winston transport:
+ *
+ * import { log } from "logger";
+ * import {TelemetryClient } from "applicationinsights";
+ * import {
+ *   ApplicationInsightsWinstonTransport,
+ *   wrapCustomTelemetryClient
+ * } from "./applicationinsights";
+ *
+ * const telemetryClient = wrapCustomTelemetryClient(
+ *   isProduction,
+ *   TelemetryClient()
+ * )();
+ *
+ * log.add(new ApplicationInsightsWinstonTransport(telemetryClient));
+ *
+ */
+import * as ApplicationInsights from "applicationinsights";
+import {
+  Data,
+  EventData
+} from "applicationinsights/out/Declarations/Contracts";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { ulid } from "ulid";
+import * as WinstonTransport from "winston-transport";
+import { IWinstonTransportInfo, IWinstonTransportLevel } from "./logger";
+
+export interface ITelemetryParams {
+  readonly operationId?: NonEmptyString;
+  readonly operationParentId?: NonEmptyString;
+}
+
+/**
+ * A TelemetryClient instance cannot be shared between track calls
+ * as custom properties and tags (ie. operationId) are part
+ * of a mutable shared state attached to the instance.
+ *
+ * This method returns a TelemetryClient with a custom
+ * TelemetryProcessor that stores tags and properties
+ * before any call to track(); in this way the returned
+ * instance of the TelemetryClient can be shared safely.
+ */
+export function wrapCustomTelemetryClient(
+  isTracingDisabled: boolean,
+  client: ApplicationInsights.TelemetryClient
+): (
+  params?: ITelemetryParams,
+  commonProperties?: Record<string, string>
+) => ApplicationInsights.TelemetryClient {
+  if (isTracingDisabled) {
+    // this won't disable manual calls to trackEvent / trackDependency
+    ApplicationInsights.Configuration.setAutoCollectConsole(false)
```

Comment:
not sure it makes sense to have this side effect here, this gets called every time we create a new client and will also change the appinsights configuration globally

Reply:
also, I don't see any call to this function in the code? where does it gets used?

Reply:
This code is now in io-functions-commons here: https://github.com/teamdigitale/io-functions-commons/blob/master/src/utils/application_insights.ts

` this gets called every time we create a new client` <-- the applicationinsights client (like the other azure clients) should be a singleton so `wrapCustomTelemetryClient` should be called once in the bootstrap phase (its purpose is to globally configure the (singleton) appinsights client).

Reply:
ops I didn't notice this was the old PR :) pls check the new one https://github.com/teamdigitale/io-backend/pull/503